x-pack/plugins/security_solution/public/management/pages/response_actions/view/response_actions_list.tsx

```diff
@@ -0,0 +1,422 @@
+                    : isCompleted
+                    ? data.wasSuccessful
+                      ? 'success'
+                      : 'danger'
+                    : 'warning'
+                }
+              >
+                <FormattedMessage
+                  id="xpack.securitySolution.responseActionsList.list.item.status"
+                  defaultMessage="{status}"
+                  values={{ status }}
+                />
+              </EuiBadge>
+            </EuiToolTip>
+          );
+        },
+      },
+      {
+        field: '',
+        align: RIGHT_ALIGNMENT as HorizontalAlignment,
+        width: '40px',
+        isExpander: true,
+        name: (
+          <EuiScreenReaderOnly>
+            <span>{UX_MESSAGES.screenReaderExpand}</span>
+          </EuiScreenReaderOnly>
+        ),
+        render: (data: ActionDetails) => {
+          return (
+            <EuiButtonIcon
+              onClick={() => toggleDetails(data)}
+              aria-label={itemIdToExpandedRowMap[data.id] ? 'Collapse' : 'Expand'}
+              iconType={itemIdToExpandedRowMap[data.id] ? 'arrowUp' : 'arrowDown'}
+            />
+          );
+        },
+      },
+    ];
+    if (hideHostNameColumn) {
+      columns.splice(3, 1);
+    }
+    return columns;
+  }, [getTestId, hideHostNameColumn, itemIdToExpandedRowMap, toggleDetails]);
+
+  // table pagination
+  const tablePagination = useMemo(() => {
+    return {
+      pageIndex: queryParams.page || 0,
+      pageSize: queryParams.pageSize || 10,
+      totalItemCount: totalItemCount || 0,
+      pageSizeOptions: MANAGEMENT_PAGE_SIZE_OPTIONS as number[],
+    };
+  }, [queryParams, totalItemCount]);
+
+  // handle onChange
+  const handleTableOnChange = useCallback(
+    ({ page: _page }: CriteriaWithPagination<ActionDetails>) => {
+      const { index, size } = _page;
+      setQueryParams((prevState) => ({
+        ...prevState,
+        page: index,
+        pageSize: size,
+      }));
+      reFetchEndpointActionList();
+    },
+    [reFetchEndpointActionList, setQueryParams]
+  );
+
+  return (
+    <AdministrationListPage
+      data-test-subj="responseActionsPage"
+      title={hideHeader ? undefined : UX_MESSAGES.pageTitle}
+    >
+      {isFetched && !totalItemCount ? (
+        <ManagementEmptyStateWrapper>
+          <EuiFlexItem data-test-subj={getTestId('empty-prompt')}>
+            <EuiEmptyPrompt
+              iconType="editorUnorderedList"
+              titleSize="s"
+              title={
+                <h2>
+                  <FormattedMessage
+                    id="xpack.securitySolution.responseActionsList.empty.title"
+                    defaultMessage="No response actions log"
+                  />
+                </h2>
+              }
+              body={
+                <p>
+                  <FormattedMessage
+                    id="xpack.securitySolution.responseActionsList.empty.body"
+                    defaultMessage="Try a different set of filters"
```

Comment:
Do we show a search bar/filter bar when there are no results?

can you include a screen capture of the empty state

Reply:
ohhh. I see. ok. But: what if there are no log entries at all? will we just be telling the user to just "keep trying different filters" knowing output will never be returned?

Normally:

- We would check the API twice - once to ensure data exists (with `page=1, perPage=1`) and the regular API call
- We first wait for the "does data exist" call:
    - if we have data, then we can display the "regular" output, even if it return 0 records
    - If no data exists, then we show an empty state with no other options